contrib/labstack/echo.v4/echotrace.go

```diff
@@ -40,9 +40,14 @@ func Middleware(opts ...Option) echo.MiddlewareFunc {
 	log.Debug("contrib/labstack/echo.v4: Configuring Middleware: %#v", cfg)
 	spanOpts := []ddtrace.StartSpanOption{
 		tracer.ServiceName(cfg.serviceName),
+	}
+	for k, v := range cfg.tags {
+		spanOpts = append(spanOpts, tracer.Tag(k, v))
+	}
+	spanOpts = append(spanOpts,
 		tracer.Tag(ext.Component, componentName),
 		tracer.Tag(ext.SpanKind, ext.SpanKindServer),
```

Comment:
Not blocking nit: as a middleware is a hot path, it benefits from reducing allocations by preallocating the slice with a size of the total number of options.

```suggestion
	spanOpts := make([]ddtrace.StartSpanOption, 0, 3+len(cfg.tags))
	spanOpts = append(spanOpts, tracer.ServiceName(cfg.serviceName))
	for k, v := range cfg.tags {
		spanOpts = append(spanOpts, tracer.Tag(k, v))
	}
	spanOpts = append(spanOpts,
		tracer.Tag(ext.Component, componentName),
		tracer.Tag(ext.SpanKind, ext.SpanKindServer),
```

Reply:
Good suggestion! I will patch it here. I don't trust the Github suggestion apply mechanism, always ends up with some non-gofmt code.